examples/src/rescue_raps/custom_trace_table.rs

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use core_utils::{collections::Vec, uninit_vector};
+use winterfell::{
+    math::{log2, FieldElement, StarkField},
+    EvaluationFrame, Matrix, Trace, TraceInfo, TraceLayout,
+};
+
+// RAP TRACE TABLE
+// ================================================================================================
+/// A concrete implementation of the [Trace] trait supporting custom RAPs.
+///
+/// This implementation supports concurrent trace generation and should be sufficient for most use
+/// cases. There are two ways to create a trace table trace.
+///
+/// First, you can use the [RapTraceTable::init()] function which takes a set of vectors as a
+/// parameter, where each vector contains values for a given column of the trace. This approach
+/// allows you to build an execution trace as you see fit, as long as it meets a basic set of
+/// requirements. These requirements are:
+///
+/// 1. Lengths of all columns in the execution trace must be the same.
+/// 2. The length of the columns must be some power of two.
+///
+/// The other approach is to use [RapTraceTable::new()] function, which takes trace width and
+/// length as parameters. This function will allocate memory for the trace, but will not fill it
+/// with data. To fill the execution trace, you can use the [fill()](RapTraceTable::fill) method,
+/// which takes two closures as parameters:
+///
+/// 1. The first closure is responsible for initializing the first state of the computation
+///    (the first row of the execution trace).
+/// 2. The second closure receives the previous state of the execution trace as input, and must
+///    update it to the next state of the computation.
+///
+/// You can also use [RapTraceTable::with_meta()] function to create a blank execution trace.
+/// This function work just like [RapTraceTable::new()] function, but also takes a metadata
+/// parameter which can be an arbitrary sequence of bytes up to 64KB in size.
+///
+/// # Concurrent trace generation
+/// For computations which consist of many small independent computations, we can generate the
+/// execution trace of the entire computation by building fragments of the trace in parallel,
+/// and then joining these fragments together.
+///
+/// For this purpose, `RapTraceTable` struct exposes [fragments()](RapTraceTable::fragments)
+/// method, which takes fragment length as a parameter, breaks the execution trace into equally
+/// sized fragments, and returns an iterator over these fragments. You can then use fragment's
+/// [fill()](RapTraceTableFragment::fill) method to fill all fragments with data in parallel.
+/// The semantics of the fragment's [RapTraceTableFragment::fill()] method are identical to the
+/// semantics of the [RapTraceTable::fill()] method.
```

Comment:
This section (and some of the above comments) are outdated. We can probably remove most of them and just have a simple description.

Reply:
Yes sorry! I've removed some others as well.